### PR TITLE
12 Strict

### PR DESCRIPTION
**Pull request by smazik02**
Sizeable code refactor to implement strict null checking. Fixed few bugs here and there where found and appropriate.

### CODE REVIEW

File: src/auth/strategy/Jwt.Strategy.ts [hunk 1]
```diff
@@ -19,11 +19,9 @@ export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
         });
     }
 
-    async validate(payload: {
-        sub: number;
-        email: string;
-    }): Promise<UserEntity> {
+    async validate(payload: { sub: number; email: string }) {
         const user = await this.userRepository.findOneBy({ id: payload.sub });
+        if (!user) return null;
```
[the remark this answers is not in the document]
Comment (Aeri4a) [reply src/auth/strategy/Jwt.Strategy.ts:24]: For a good practice actually you could replace from line 23 to line 26 with one line return. You could use support map dto function (which also you can use in whole app). Think about it.

File: src/config/app.config.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { registerAs } from '@nestjs/config';
 
 export default registerAs('config', () => ({
-    port: parseInt(process.env.PORT, 10) || 5000,
+    port: parseInt(<string>process.env.PORT, 10) || 5000,
```
Comment (Aeri4a) [inline src/config/app.config.ts:4]: Use nullish operator instead of assertion

File: src/config/database.config.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { registerAs } from '@nestjs/config';
 export default registerAs('database', () => ({
     type: process.env.DB_TYPE || 'mysql',
     host: process.env.DB_HOSTNAME || 'localhost',
-    port: parseInt(process.env.DB_PORT, 10) || 3306,
+    port: parseInt(<string>process.env.DB_PORT, 10) || 3306,
```
Comment (Aeri4a) [inline src/config/database.config.ts:6]: same

File: src/config/orm.config.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { DataSource } from 'typeorm';
 const ormConfig = new DataSource({
     type: 'mysql',
     host: process.env.DB_HOSTNAME,
-    port: parseInt(process.env.DB_PORT, 10),
+    port: parseInt(<string>process.env.DB_PORT, 10),
```
Comment (Aeri4a) [inline src/config/orm.config.ts:6]: same

File: src/resource/event/event.controller.ts [hunk 5]
```diff
@@ -74,15 +75,20 @@ export class EventController {
     @Get(':id')
     @GetEventByIdApi()
     async getEventById(@Param('id') eventId: number): Promise<ReturnEventDTO> {
-        return await this.eventService.fetchById(eventId);
+        const event = await this.eventService.fetchById(eventId);
+        if (!event) throw new NotFoundException('Event not found');
+        return event;
```
Comment (Aeri4a) [inline src/resource/event/event.controller.ts:80]: then, catch?

File: src/resource/event/event.service.ts [hunk 6]
```diff
@@ -33,13 +33,11 @@ export class EventService {
         private readonly exceptionRepository: Repository<EventExceptionEntity>,
     ) {}
 
-    async fetchById(eventId: number): Promise<ReturnEventDTO> {
-        const event = await this.eventRepository.findOne({
+    async fetchById(eventId: number): Promise<ReturnEventDTO | null> {
```
Comment (Aeri4a) [inline src/resource/event/event.service.ts:36]: The thing is that we don't need every time to have return type of a method explicitly. If you hover at function and see the correct type from a implicit typing, then it is all right. It should be done in the whole app.

File: src/resource/note/note.controller.ts [hunk 7]
```diff
@@ -34,13 +36,22 @@ export class NoteController {
...
-    addNote(@GetUser() user: UserEntity, @Body() noteDto: NoteDTO): Promise<NoteDTO> {
-        return this.notesService.add(user, noteDto);
+    async addNote(
+        @GetUser() user: UserEntity,
+        @Body() noteDto: NoteDTO,
+    ): Promise<NoteDTO> {
+        const note = await this.notesService.add(user, noteDto);
+        if (!note)
+            throw new HttpException('User not found', HttpStatus.NOT_FOUND);
+        return note;
```
Comment (Aeri4a) [inline src/resource/note/note.controller.ts:46]: No need for explicit + then catch?

File: src/resource/task/task.controller.ts [hunk 8]
```diff
@@ -36,9 +36,7 @@ export class TaskController {
 
     @Get()
     @GetUserTasksApi()
-    async getUserTasks(
-        @GetUser() user: UserEntity,
-    ): Promise<ReturnTaskDTO[]> {
+    async getUserTasks(@GetUser() user: UserEntity): Promise<ReturnTaskDTO[]> {
```
Comment (Aeri4a) [inline src/resource/task/task.controller.ts:39]: Explicit type

File: src/resource/task/task.service.ts [hunk 9]
```diff
@@ -12,7 +12,7 @@ export class TaskService {
         private readonly taskRepository: Repository<TaskEntity>,
     ) {}
 
-    async fetchById(user: UserEntity, id: number): Promise<TaskEntity> {
+    async fetchById(user: UserEntity, id: number): Promise<TaskEntity | null> {
```
Comment (Aeri4a) [inline src/resource/task/task.service.ts:15]: same

File: src/resource/task/task.service.ts [hunk 10]
```diff
@@ -35,13 +35,13 @@ export class TaskService {
     async edit(
         user: UserEntity,
         editTask: EditTaskDTO,
-    ): Promise<ReturnTaskDTO> {
+    ): Promise<ReturnTaskDTO | null> {
```
Comment (Aeri4a) [inline src/resource/task/task.service.ts:38]: same

File: src/resource/task/task.service.ts [hunk 11]
```diff
@@ -35,13 +35,13 @@ export class TaskService {
...
         const task = await this.taskRepository.findOneBy({
             id: editTask.id,
             user,
         });
         if (!task) {
-            throw new NotFoundException('Task not found');
+            return null;
         }
         return await this.taskRepository.save({ ...editTask });
```
Comment (Aeri4a) [inline src/resource/task/task.service.ts:46]: Format as I mentioned

File: src/resource/user/user.controller.ts [hunk 12]
```diff
@@ -27,27 +28,30 @@ export class UserController {
     @Get('me')
     @GetMeApi()
     getMe(@GetUser() user: UserEntity): ReturnUserDTO {
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const { hash, ...result } = user;
-        return result;
+        delete user.hash;
+        return user;
```
Comment (Aeri4a) [inline src/resource/user/user.controller.ts:32]: Can we do it in better way?

File: src/resource/user/user.controller.ts [hunk 13]
```diff
@@ -27,27 +28,30 @@ export class UserController {
...
     @EditMeApi()
-    editMe(
+    async editMe(
         @GetUser('id') userId: number,
         @Body() editUserDto: EditUserDTO,
     ): Promise<ReturnUserDTO> {
-        return this.userService.editMe(userId, editUserDto);
+        const user = await this.userService.editMe(userId, editUserDto);
+        if (!user) throw new NotFoundException('User not found');
+        return user;
```
Comment (Aeri4a) [inline src/resource/user/user.controller.ts:43]: then catch

File: src/resource/user/user.controller.ts [hunk 14]
```diff
@@ -27,27 +28,30 @@ export class UserController {
...
     @GetUserByIdApi()
-    getUserById(
+    async getUserById(
         @Param('id', ParseIntPipe) userId: number,
     ): Promise<ReturnUserDTO> {
-        return this.userService.getUserById(userId);
+        const user = await this.userService.getUserById(userId);
+        if (!user) throw new NotFoundException('User not found');
+        return user;
     }
```
Comment (Aeri4a) [inline src/resource/user/user.controller.ts:56]: then catch

File: src/resource/user/user.service.ts [hunk 15]
```diff
@@ -14,9 +14,12 @@ export class UserService {
     async editMe(
         userId: number,
         editUserDto: EditUserDTO,
-    ): Promise<ReturnUserDTO> {
-        const user = await this.userRepository.findOneBy({ id: userId });
-        if (!user) throw new NotFoundException('User not found');
+    ): Promise<ReturnUserDTO | null> {
```
Comment (Aeri4a) [inline src/resource/user/user.service.ts:17]: explicit type not necessary

File: src/resource/user/user.service.ts [hunk 16]
```diff
@@ -25,16 +28,14 @@ export class UserService {
...
-    async getUserById(userId: number): Promise<ReturnUserDTO> {
+    async getUserById(userId: number): Promise<ReturnUserDTO | null> {
         const user = await this.userRepository.findOneBy({ id: userId });
-        if (!user) throw new NotFoundException('User not found');
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const { hash, ...result } = user;
-        return result;
+        if (!user) return null;
+        delete user.hash;
+        return user;
```
Comment (Aeri4a) [inline src/resource/user/user.service.ts:39]: Other way

File: tsconfig.json
Comment (Aeri4a) [inline tsconfig.json:1]: Make full strict for the whole project

**Review by Aeri4a — CHANGES_REQUESTED** [review]

File: src/config/app.config.ts (hunk 2 above)
Comment (smazik02) [reply src/config/app.config.ts]: Done

File: src/config/database.config.ts (hunk 3 above)
Comment (smazik02) [reply src/config/database.config.ts]: Done

File: src/config/orm.config.ts (hunk 4 above)
Comment (smazik02) [reply src/config/orm.config.ts]: Done

File: tsconfig.json
Comment (smazik02) [reply tsconfig.json:1]: Done

File: src/auth/strategy/Jwt.Strategy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (smazik02) [reply src/auth/strategy/Jwt.Strategy.ts]: Implemented dto mapping, made the strategy and all controllers much more concise

File: src/resource/event/event.service.ts (hunk 6 above)
Comment (smazik02) [reply src/resource/event/event.service.ts]: Done

File: src/database/entities/event.entity.ts [hunk 17]
```diff
@@ -56,4 +56,30 @@ export class EventEntity {
         cascade: true,
     })
     eventExceptions: EventExceptionEntity[];
+
```
Comment (Aeri4a) [inline src/database/entities/event.entity.ts:59]: Find out if there is a way to skip constructors

File: src/resource/event/event.service.ts [hunk 18]
```diff
@@ -33,25 +34,24 @@ export class EventService {
...
     ) {}
 
-    async fetchById(eventId: number): Promise<ReturnEventDTO | null> {
-        return await this.eventRepository.findOne({
+    async fetchById(eventId: number) {
+        const event = await this.eventRepository.findOne({
             where: { id: eventId },
             relations: { recurrencePattern: true },
         });
+        if (!event) throw new NotFoundException('Event not found');
```
Comment (Aeri4a) [inline src/resource/event/event.service.ts:42]: findOneOrThrow -> remove if and throw new error in catch from this chain

File: src/resource/note/dto/note.dto.ts [hunk 19]
```diff
@@ -1,28 +1,51 @@
-import { IsOptional, IsString } from 'class-validator';
 import { ApiProperty } from '@nestjs/swagger';
+import { Expose } from 'class-transformer';
+import { IsOptional, IsString } from 'class-validator';
```
Comment (Aeri4a) [inline src/resource/note/dto/note.dto.ts:4]: Remove constructors from dto (u know how) and remove expose, and all mappings to entity in dto, dto in dto should be only allowed.

**Review by Aeri4a — APPROVED** [review]

**Pull request merged**